### PR TITLE
Display the rewards icon when the rewards panel is requested

### PR DESCRIPTION
**Pull request by zenparsing**
Resolves https://github.com/brave/brave-browser/issues/19211

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

*TODO*

### CODE REVIEW

File: browser/ui/views/brave_actions/brave_actions_container_browsertest.cc [hunk 1]
```diff
@@ -99,3 +99,22 @@ IN_PROC_BROWSER_TEST_F(BraveActionsContainerTest,
...
+
+  // Simulate an action from the brave actions API to open the rewards panel.
+  extensions::BraveActionAPI::Observer* action_observer = brave_actions_;
+  action_observer->OnBraveActionShouldTrigger(brave_rewards_extension_id,
+                                              nullptr);
+
+  base::RunLoop().RunUntilIdle();
+
+  // Rewards action should be shown while popup is open.
+  CheckBraveRewardsActionShown(true);
```
Comment (simonhong) [inline browser/ui/views/brave_actions/brave_actions_container_browsertest.cc:119]: Can we check rewards action is hidden again after popup is closed?

File: browser/ui/views/brave_actions/brave_actions_container.cc [hunk 2]
```diff
@@ -279,16 +232,25 @@ void BraveActionsContainer::RemoveAction(const std::string& id) {
...
-void BraveActionsContainer::ShowAction(const std::string& id, bool show) {
-  if (show != IsActionShown(id))
-    show ? AddAction(id) : RemoveAction(id);
+void BraveActionsContainer::UpdateActionVisibility(const std::string& id) {
+  if (views::Button* button = GetActionButton(id)) {
+    bool should_show = ShouldShowAction(id);
+    if (button->GetVisible() != should_show) {
+      button->SetVisible(should_show);
+      Update();
+      PreferredSizeChanged();
```
Comment (simonhong) [inline browser/ui/views/brave_actions/brave_actions_container.cc:241]: This is redundant call. - When child view's visibility is changed, this container(parent) will do re-layout.

**Review by simonhong — APPROVED** [review]
++ with some comments.

File: browser/ui/views/brave_actions/brave_actions_container_browsertest.cc (hunk 1 above)
Comment (zenparsing) [reply browser/ui/views/brave_actions/brave_actions_container_browsertest.cc:119]: Good idea - thanks!

File: browser/ui/views/brave_actions/brave_actions_container.cc (hunk 2 above)
Comment (zenparsing) [reply browser/ui/views/brave_actions/brave_actions_container.cc]: Fixed.

**Pull request merged**